### PR TITLE
Better explain that form types should be unique in the application

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | all |
| Fixed tickets | #4960 |

### CODE REVIEW

File: book/forms.rst [hunk 1]
```diff
@@ -1057,9 +1057,15 @@ that will house the logic for building the task form::
...
 
-This new class contains all the directions needed to create the task form
-(note that the ``getName()`` method should return a unique identifier for this
-form "type"). It can be used to quickly build a form object in the controller::
+.. caution::
+
+    The ``getName()`` method returns the identifier of this form "type". These
+    identifiers must be unique in the application and, unless you want to override
+    a built-in type, they should be different from the default Symfony types.
+    Consider prefixing your types with ``app_`` to avoid identifier collisions.
```
Comment (stof) [inline book/forms.rst]: should this become a best practice ?

File: book/forms.rst [hunk 2]
```diff
@@ -1057,9 +1057,15 @@ that will house the logic for building the task form::
...
     }
 
-This new class contains all the directions needed to create the task form
-(note that the ``getName()`` method should return a unique identifier for this
-form "type"). It can be used to quickly build a form object in the controller::
+.. caution::
+
+    The ``getName()`` method returns the identifier of this form "type". These
+    identifiers must be unique in the application and, unless you want to override
+    a built-in type, they should be different from the default Symfony types.
```
Comment (stof) [inline book/forms.rst]: or from any type provided in a third-party bundle, not only in Symfony

File: book/forms.rst [hunk 3]
```diff
@@ -1057,9 +1057,15 @@ that will house the logic for building the task form::
...
+    Consider prefixing your types with ``app_`` to avoid identifier collisions.
+
+This new class contains all the directions needed to create the task form. It can
+be used to quickly build a form object in the controller:
```
Comment (xabbuh) [inline book/forms.rst]: We need to colons here. Otherwise, the following code block won't be recognized.

**Comment by javiereguiluz** [comment]
Thanks for your review. Everything has been fixed/updated and I've created a separate issue (https://github.com/symfony/symfony-docs/issues/5077) to discuss about introducing a new best practice for Symfony forms.

File: book/forms.rst [hunk 4]
```diff
@@ -1057,9 +1057,16 @@ that will house the logic for building the task form::
...
-This new class contains all the directions needed to create the task form
-(note that the ``getName()`` method should return a unique identifier for this
-form "type"). It can be used to quickly build a form object in the controller::
+.. caution::
+
+    The ``getName()`` method returns the identifier of this form "type". These
+    identifiers must be unique in the application. Unless you want to override
+    a built-in type, they should be different from the default Symfony types
+    and from any type defined by a third-party bundle installed in you application.
+    Consider prefixing your types with ``app_`` to avoid identifier collisions.
```
Comment (xabbuh) [inline book/forms.rst]: In the best practices, we use `app.` (dot instead of underscore) for service names. Should we follow this convention here too?

Comment (javiereguiluz) [inline book/forms.rst]: I agree with you. I've just updated the PR with your suggestion.

File: book/forms.rst [hunk 5]
```diff
@@ -1057,9 +1057,16 @@ that will house the logic for building the task form::
...
-This new class contains all the directions needed to create the task form
-(note that the ``getName()`` method should return a unique identifier for this
-form "type"). It can be used to quickly build a form object in the controller::
+.. caution::
+
+    The ``getName()`` method returns the identifier of this form "type". These
+    identifiers must be unique in the application. Unless you want to override
+    a built-in type, they should be different from the default Symfony types
+    and from any type defined by a third-party bundle installed in you application.
+    Consider prefixing your types with ``app.`` to avoid identifier collisions.
```
Comment (xabbuh) [inline book/forms.rst]: Sorry, Javier, we also need to switch back to `app_` here as @stof explained in #5077.

Comment (javiereguiluz) [inline book/forms.rst]: You are right. It's done in 0eb149b. Thanks.

File: book/forms.rst [hunk 6]
```diff
@@ -1057,9 +1057,16 @@ that will house the logic for building the task form::
...
 
-This new class contains all the directions needed to create the task form
-(note that the ``getName()`` method should return a unique identifier for this
-form "type"). It can be used to quickly build a form object in the controller::
+.. caution::
+
+    The ``getName()`` method returns the identifier of this form "type". These
+    identifiers must be unique in the application. Unless you want to override
+    a built-in type, they should be different from the default Symfony types
+    and from any type defined by a third-party bundle installed in you application.
```
Comment (xabbuh) [inline book/forms.rst]: your

**Comment by weaverryan** [comment]
Thanks Javier! I don't like *needing* this warning, but it makes sense. It does beg the question though - should we prefix all of these with app_ in the docs?

**Comment by weaverryan** [comment]
Ha, yes I see that's exactly what you're asking in #5077 :)

**Pull request merged**